gnomad/variant_qc/training.py

```diff
@@ -0,0 +1,130 @@
+import logging
+from typing import List, Optional, Tuple
+
+import hail as hl
+from pprint import pformat
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def sample_training_examples(
+    ht: hl.Table,
+    tp_expr: hl.BooleanExpression,
+    fp_expr: hl.BooleanExpression,
+    fp_to_tp: float = 1.0,
+    test_expr: Optional[hl.expr.BooleanExpression] = None,
+) -> hl.Table:
+    """
+    Returns a Table of all positive and negative training examples in `ht` with an annotation for those that should be
+    used for training given a true positive (TP) to false positive (FP) ratio.
+
+    The returned Table has the following annotations:
+        - train: indicates if the variant should be used for training. The only variants labeled as False will be
```

Comment:
There are two other cases for variants to be excluded for training:
- variants that are `True` for both `tp_expr` and `fp_expr` will also be `False`
- variants that are pruned out to obtain the desired `fp_to_tp` ratio.